week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -1,35 +1,38 @@
 /* 
 Password Validation
-
 Write a program that should check if each password in an array 
 contains a valid password (see below for password criterias) and return 
 new array with true or false booleans.
-
 Passwords must 
 - Have at least 5 characters.
 - Have English uppercase letters (A-Z)
 - Have English lowercase letters (a-z)
 - Have numbers (0-9)
 - Have non-alphanumeric symbols ("!", "#", "$", "%", ".")
-
 Passwords must not be any previous password in the passwords array. 
-
 Example 1:
 PreviousPassword = ["fhD8!yrjj", "ttkTu.wer3", "dvyyeyY!5", "qwbfj76%", "tytT3729."];
-
 Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
+*/ 
 
-*/
+ 
+function validatePasswords(arr) {
 
-function validatePasswords(passwords) {
 
-}
+    let validatePass = /(?=.{5,})(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!#$%.])/
```

Comment:
Wow, this is a complicated regex! Do you feel like you understand what is happening here? If you do, could you explain how it works?

If not, I would recommend reading up on Regular Expressions (aka *regexes*). [This looks like a good place to start reading](https://codeburst.io/javascript-learn-regular-expressions-for-beginners-bb6107015d91). To see an example, Min Ko's homework has some really nice usage of regexes: https://github.com/CodeYourFuture/js-exercises/pull/499/files#diff-548cf5a2c7a92b75e67ed7cf8afdbbfbR26-R29 (scroll down to the bottom, it should be highlighted in yellow).

Reply:
I initially tested the code on https://regex101.com/ and thought it would be easy to break down by following each parenthesis.  Min's code looks more clear and it would be easier for people to read it and spot if there are mistakes along the way. Thank you for the resource